docs/mesh/packing.md

```diff
@@ -0,0 +1,65 @@
+# Packing Variables Accross the Entire Mesh
+
+`Kokkos` kernel launches have a 6 microsecond overhead. For small
+kernels that perform little work, this can be a perforamnce bottleneck
+when each kernel is launched per meshblock. Parthenon therefore
+provides the capability to combine variables into a single data
+structure that spans all meshblocks, the `MeshPack`. 
+
+## Creating a Mesh Pack
+
+There are two methods for creating mesh packs, which are analogous to the `VariablePack` and `VariableFluxPack` available in [containers](../interface/containers.md).
+```C++
+template <typename... Args>
+auto PackVariablesOnMesh(Mesh *pmesh, const std::string &container_name,
+                         Args &&... args)
+```
+and
+```C++
+auto PackVariablesAndFluxesOnMesh(Mesh *pmesh, const std::string &container_name,
+                                  Args &&... args)
+```
+The former packs only the variables, the latter packs in the variables and associated flux vectors.
+
+The variatic arguments take exactly the same arguments as
+`container.PackVariables` and `container.PackVariablesAndFluxes`. You
+can pass in a metadata vector, a vector of names, and optionally IDs
+and a map from names to indices. See
+[here](../interface/containers.md) for more details.
+
+## Data Layout
+
+The Mesh Pack is indexable as a five-dimensional `Kokkos` view. The
+slowest moving index indexes into a 4-dimensional `VariablePack`. The
+next slowest indexes into a variable. The fastest three index into the
+cells on a meshblock.
+
+For example:
+```C++
+// pack all variables on the base container accross the whole mesh
+auto meshpack = PackVariablesAndFluxes(pmesh, "base");
+auto variablepack = meshpack(m); // Indexes into the m'th meshblock
+auto auto var = mesh(m,l); // Indexes into the k'th variable on the m'th MB
```

Comment:
The elusive `auto auto`.

Reply:
Is `mesh` supposed to be `meshpack` here?

Reply:
good catches thanks